src/App.tsx

```diff
@@ -1,56 +1,90 @@
+      .includes(appliedQuery.toLowerCase()));
+  }, [appliedQuery]);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const inputField = useRef<HTMLInputElement>(null);
+
+  const showSugesstion = (appliedQuery === query) && isFocused;
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? `${selectedPerson.name} (${selectedPerson.born} = ${selectedPerson.died})`
+          : 'No selected person'}
       </h1>
 
       <div className="dropdown is-active">
         <div className="dropdown-trigger">
           <input
             type="text"
+            id="input"
+            ref={inputField}
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={handleQueryChange}
+            onFocus={() => setIsFocused(true)}
+            onBlur={() => setIsFocused(false)}
           />
         </div>
 
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
+        {showSugesstion && (
+          <div className="dropdown-menu" role="menu">
+            <div className="dropdown-content">
+              {visiblePerson.length === 0
+                ? (
+                  <div className="dropdown-item">
+                    <p className="has-text-danger">
+                      No matching suggestions
+                    </p>
+                  </div>
+                )
+                : (
+                  visiblePerson.map((person: Person) => (
+                    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+                    <div
+                      role="tooltip"
+                      className="dropdown-item"
+                      key={peopleFromServer.indexOf(person)}
```

Comment:
```suggestion
                      key={person.slug}
```
in person object exists unique field. so no need use index or indexOf